src/main/java/org/shanerx/tradeshop/bitrade/BiShopCreateEventListener.java

```diff
@@ -44,51 +45,50 @@ public BiShopCreateEventListener(TradeShop instance) {
 	}
 	
 	@SuppressWarnings("deprecation")
-	@EventHandler
+	@EventHandler(priority = EventPriority.HIGH)
 	public void onSignChange(SignChangeEvent event) throws InterruptedException {
-		//	BlockState state = event.getBlock().getState();
 		Player player = event.getPlayer();
 		Sign s = (Sign) event.getBlock().getState();
-		if (!(event.getLine(0).equalsIgnoreCase("[BiTrade]"))) {
+		
+		if (!event.getLine(0).equalsIgnoreCase("[BiTrade]")) {
+			return;
+			
+		} else if (!player.hasPermission(getCreatePerm())) {
+			s.setLine(0, "");
+			s.update();
+			s.setLine(1, "");
+			s.update();
+			s.setLine(2, "");
+			s.update();
+			s.setLine(3, "");
+			s.update();
+			player.sendMessage(colorize(getPrefix() + plugin.getMessages().getString("no-ts-create-permission")));
 			return;
 		}
-		
-		if (!player.hasPermission(getCreatePerm())) {
-            s.setLine(0, "");
-            s.update();
-            s.setLine(1, "");
-            s.update();
-            s.setLine(2, "");
-            s.update();
-            s.setLine(3, "");
-            s.update();
-            player.sendMessage(colorize(getPrefix() + plugin.getMessages().getString("no-ts-create-permission")));
-            return;
-        }
 		Block chest;
 		
 		try {
-		    chest = findShopChest(s.getBlock());
-		} catch(Exception e) {
-		    event.setLine(0, ChatColor.DARK_RED + "[BiTrade]");
-            event.setLine(1, "");
-            event.setLine(2, "");
-            event.setLine(3, "");
-            player.sendMessage(colorize(getPrefix() + plugin.getMessages().getString("no-chest")));
-            return;
+			chest = findShopChest(s.getBlock());
+		} catch (Exception e) {
+			event.setLine(0, ChatColor.DARK_RED + "[BiTrade]");
+			event.setLine(1, "");
+			event.setLine(2, "");
+			event.setLine(3, "");
+			player.sendMessage(colorize(getPrefix() + plugin.getMessages().getString("no-chest")));
+			return;
 		}
 		
-
-        if (!plugin.getAllowedInventories().contains(chest.getType())) {
-            event.setLine(0, ChatColor.DARK_RED + "[BiTrade]");
-            event.setLine(1, "");
-            event.setLine(2, "");
-            event.setLine(3, "");
-            player.sendMessage(colorize(getPrefix() + plugin.getMessages().getString("no-chest")));
-            return;
-        }
-        
-		boolean signIsValid = true; // If this is true, the information on the sign is valid!
+		
+		if (!plugin.getAllowedInventories().contains(chest.getType())) {
```

Comment:
maybe we should create a method in utils to set the lines... I'll add it to the TODO.